src/Chapter3.hs

```diff
@@ -1007,6 +1157,147 @@ contestants, and write a function that decides the outcome of a fight!
 -}
 
 
+data FightResult = FirstFighterFled 
+                 | SecondFighterFled
+                 | FirstFighterDefeated
+                 | SecondFighterDefeated
+                 | Draw
+                deriving Show
+
+data Knight = Knight
+    { knightHealth    :: Health
+    , knightAttack    :: Attack
+    , knightDefense   :: Defense
+    } | FightingKnight 
+    { fknightHealth               :: Health
+    , fknightMaxHealth            :: Health
+    , fknightAttack               :: Attack
+    , fknightDefense              :: Defense
+    , fknightRoundsBoostedDefense :: Int
+    }
+    deriving (Show)
```

Comment:
Interesting data type :thinking: 

Though, usually records inside sum types are discouraged because they result in partial functions. `knightHealth` is actually a function of type `Knight -> Health`, but only one constructor of the `Knight` data type has field named exactly `knightHealth`